lib/api/controller/security.js

```diff
@@ -1159,32 +1165,51 @@ class SecurityController extends NativeController {
    * @returns {Promise}
    * @private
    */
-  async _persistUser(request, profileIds, content) {
-    const id = this.getId(request, ifMissingEnum.GENERATE);
+  async _persistUser(request, profileIds, content, humanReadableId=true) {
+    const generator = humanReadableId ? nameGenerator : uuidv4;
     const credentials = this.getBodyObject(request, 'credentials', {});
     const strategies = Object.keys(credentials);
-
+    
+    let id = null;
+    let alreadExists = false;
     // Early checks before the user is created
-    if (id) {
-      for (const strategy of strategies) {
-        if (!global.kuzzle.pluginsManager.listStrategies().includes(strategy)) {
-          throw kerror.get(
-            'security',
-            'credentials',
-            'unknown_strategy',
-            strategy);
+    do {
+      let generated = false;
+      id = this.getId(
+        request,
+        ifMissingEnum.GENERATE,
+        () => {
+          generated = true;
+          return `kuid-${generator()}`;
         }
-
-        const exists = this.getStrategyMethod(strategy, 'exists');
-        if (await exists(request, id, strategy)) {
-          throw kerror.get(
-            'security',
-            'credentials',
-            'database_inconsistency',
-            id);
+      );
+
+      if (id) {
```

Comment:
I don't understand why it should be checked that id is not undefined. It will never be the case according to:
```js
      id = this.getId(
        request,
        ifMissingEnum.GENERATE,
        () => {
          generated = true;
          return `kuid-${generator()}`;
        }
      );
```

Reply:
You are right, this check is not useful, removed